tests/notnull_tests.cpp

```diff
@@ -95,6 +145,104 @@ SUITE(NotNullTests)
         int* q = nullptr;
         CHECK_THROW(p = q, fail_fast);
     }
+
+    TEST(TestNotNullRawPointerComparison)
+    {
+        int x = 1;
+        const int y = 2;
+        auto p1 = &x;
+        auto p2 = &y;
+
+        using NotNull1 = not_null<decltype(p1)>;
+        using NotNull2 = not_null<decltype(p2)>;
+
+        CHECK((NotNull1(p1) == NotNull1(p1)) == true);
+        CHECK((NotNull1(p1) == NotNull2(p2)) == false);
+
+        CHECK((NotNull1(p1) != NotNull1(p1)) == false);
+        CHECK((NotNull1(p1) != NotNull2(p2)) == true);
+
+        CHECK((NotNull1(p1) < NotNull1(p1)) == false);
+        CHECK((NotNull1(p1) < NotNull2(p2)) == (p1 < p2));
```

Comment:
The result of comparing `&x` and `&y` with `<`, '>`, `<=`, and `>=` is unspecified, and hence might not even be deterministic. The standard allows you to evaluate this condition, but says nothing about what its value will be.